trikot-streams/streams/src/commonMain/kotlin/com/mirego/trikot/streams/cancellable/CancellableManagerProvider.kt

```diff
@@ -1,20 +1,29 @@
 package com.mirego.trikot.streams.cancellable
 
 import com.mirego.trikot.foundation.concurrent.AtomicReference
+import com.mirego.trikot.foundation.concurrent.dispatchQueue.SynchronousSerialQueue
 
 class CancellableManagerProvider : Cancellable {
-    private val cancellableManager = CancellableManager()
+    private val serialQueue = SynchronousSerialQueue()
     private val internalCancellableManagerRef = AtomicReference(CancellableManager())
+    private val isCancelled = AtomicReference(false)
 
     fun cancelPreviousAndCreate(): CancellableManager {
-        internalCancellableManagerRef.value.cancel()
-        return CancellableManager().also {
-            internalCancellableManagerRef.setOrThrow(internalCancellableManagerRef.value, it)
-            cancellableManager.add(it)
+        return CancellableManager().also { cancellableManager ->
+            internalCancellableManagerRef.getAndSet(cancellableManager).cancel()
+            serialQueue.dispatch {
```

Comment:
The usage of the serialQueue is to ensure sequential cancellation of currently returned ref

Reply:
This essentially mimicks the CancellableManager internal implementation to be on the safe side